app/controllers/chapters_controller.rb

```diff
@@ -70,8 +70,13 @@ def show
       # update the history.
       Reading.update_or_create(@work, current_user) if current_user
 
-      # TEMPORARY hack-like thing to fix the fact that chaptered works weren't hit-counted or added to history at all
-      if chapter_position == 0
+      # TEMPORARY hack-like thing to fix the fact that chaptered works weren't hit-counted
+      # or added to history at all. Increases hit count if accessing the first chapter or
+      # if accessing the most recent chapter when you're not already on the work (e.g. if
+      # you followed the recent chapter link or came from a subscription email).
+      if chapter_position == 0 || chapter_position == (@work.number_of_posted_chapters - 1) &&
+                                  (request.referer.nil? ||
+                                  !request.referer.match(/#{work_path(@work)}/))
```

Comment:
It might be good to change this regex to something like `/#{work_path(@work)}\b/` to avoid false positives (false negatives?) when one work's ID is a prefix of another's. It'd probably be pretty rare to have a work (or its comments) link to the latest chapter of a work whose ID is a prefix, but it might be worth trying to prevent errors in case it does happen.

The word boundary character should allow a regex like `/works/1\b` to match `/works/1`, `/works/1/chapter/1`, and `/works/1?view_full_work=true`, but not `/works/100`.